### DISCUSSION TITLE
Annotated syntax is too long

### DISCUSSION THREAD

**Discussion by vitalik** [fastapi/fastapi#10055] [Questions] [answered]
Introduction of `Annotated` params made function params more logical, but on the other hand longer/more verbose:

```Python
@api.post("/form")
def some_form(
    x: Annotated[int, Form()],
    y: Annotated[int, Form()],
    z: Annotated[LaxInt, Form()],
):
   ...

```

when you look at that you mostly focus on the "annotated" word, while the actual marker is harder to spot (just cognitively)

Why not make params act like Annotated without the "annotated" word:

```Python
@api.post("/form")
def some_form(
    x: Form[int],
    y: Form[int],
    z: Form[LaxInt],
):
   ...
```

this looks way more clear and backward compatible with `y: Annotated[xxx, Form]`  and  `y: xxx = Form(...)`

the implementation is actually very simple

```Python
class ParamShortcut:
    def __init__(self, base_func: Callable) -> None:  # type: ignore
        self._base_func = base_func

    def __call__(self, *args: Any, **kwargs: Any) -> Any:
        return self._base_func(*args, **kwargs)

    def __getitem__(self, args: Any) -> Any:
        if isinstance(args, tuple):
            return Annotated[args[0], self._base_func(**args[1])]
        return Annotated[args, self._base_func()]
...

Body = ParamShortcut(param_functions.Body)
Cookie = ParamShortcut(param_functions.Cookie)
Depends = ParamShortcut(param_functions.Depends)
...
``` 

implemented here #9941

**Comment by Kludex** [comment, marked as answer]
- Closing this as per https://github.com/tiangolo/fastapi/pull/9941#issuecomment-1734431608.
